documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc

```diff
@@ -0,0 +1,82 @@
+[id='rest_v2_create_protobuf']
+= Creating Protobuf Schemas
+Create Protobuf schemas across {brandname} clusters with `POST` requests that
+include the content of a protobuf file in the payload.
+
+[source,options="nowrap",subs=attributes+]
+----
+POST /rest/v2/protobuf/{schemaName}
+----
+
+If the schema already exists, {brandname} returns `CONFLICT`. If the schema is
+not valid, {brandname} adds the schema and returns `BAD_REQUEST`.
+
+{brandname} responds with the schema name and any errors.
+
+[source,json,options="nowrap",subs=attributes+]
+----
+include::json_examples/rest_protobuf_create_update_schema_response.json[]
+----
+
+* `name` is the name of the Protobuf schema.
+* `error` is `null` for valid Protobuf schemas. If {brandname} cannot successfully validate the schema, it returns errors.
+
+[id='rest_v2_get_protobuf']
+= Reading Protobuf Schemas
+Retrieve Protobuf schema from {brandname} with `GET` requests.
+
+[source,options="nowrap",subs=attributes+]
+----
+GET /rest/v2/protobuf/{schemaName}
+----
+
+[id='rest_v2_update_protobuf']
+= Updating Protobuf Schemas
+Modify Protobuf schemas with `PUT` requests that include the content of a
+protobuf file in the payload.
+
+[source,options="nowrap",subs=attributes+]
+----
+PUT /rest/v2/protobuf/{schemaName}
+----
+
+If the schema already exists, {brandname} returns `CONFLICT`. If the schema is
+not valid, {brandname} adds the schema and returns `BAD_REQUEST`.
+
+{brandname} responds with the schema name and any errors.
+
+[source,json,options="nowrap",subs=attributes+]
+----
+include::json_examples/rest_protobuf_create_update_schema_response.json[]
+----
+
+* `name` is the name of the Protobuf schema.
+* `error` is `null` for valid Protobuf schemas. If {brandname} cannot successfully validate the schema, it returns errors.
+
+[id='rest_v2_delete_protobuf']
+= Deleting Protobuf Schemas
+Remove Protobuf schemas from {brandname} clusters with `DELETE` requests.
+
+[source,options="nowrap",subs=attributes+]
+----
+DELETE /rest/v2/protobuf/{schemaName}
```

Comment:
There seems to be a lot of duplication in this doc. Or maybe I am missing something?

Reply:
@wburns thanks for commenting on the duplication. I felt it was too repetitive myself but had some doubts and needed some time to think about it. I'll add another commit with some suggestions to reduce the duplication. it's probably too distracting.